business_product_location/models/business_product_location.py

```diff
@@ -0,0 +1,192 @@
+#
+##############################################################################
+
+from openerp import api, fields, models
+import openerp.addons.decimal_precision as dp
+from openerp.tools.translate import _
+from openerp.tools.float_utils import float_compare
+
+
+class BusinessProductLocation(models.Model):
+    _name = 'business.product.location'
+
+    name = fields.Char(string='Name', required=True)
+    product_ids = fields.One2many('business.product.line',
+                                  'business_product_location_id',
+                                  string='Products')
+    location_ids = fields.One2many('stock.location',
+                                   'business_usage_id',
+                                   string='Locations')
+
+
+class BusinessProductLine(models.Model):
+    _name = 'business.product.line'
+    _rec_name = "product_id"
+
+    @api.model
+    def _default_product_uom_id(self):
+        return self.env.ref('product.product_uom_unit')
+
+    product_id = fields.Many2one('product.product', string='Product',
+                                 required=True)
+    product_qty = fields.Float(
+        'Product Quantity', required=True,
+        digits_compute=dp.get_precision('Product Unit of Measure'))
+    product_uom_id = fields.Many2one('product.uom', 'Product Unit of Measure',
+                                     required=True,
+                                     default=_default_product_uom_id)
+    business_product_location_id = fields.Many2one(
+        'business.product.location', 'Parent business product location',
+        required=True)
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        """ Change UoM if product_id changes
+        """
+        if self.product_id:
+            self.product_uom_id = self.product_id.uom_id
+
+    @api.onchange('product_uom_id')
+    def _onchange_product_uom_id(self):
+        """ Check the selected UoM with the product UoM
+        """
+        res = {}
+        if self.product_id and self.product_uom_id:
+            if self.product_id.uom_id.category_id.id != \
+                    self.product_uom_id.category_id.id:
+                res['warning'] = {
+                    'title': _('Warning'),
+                    'message': _('The Product Unit of Measure you chose '
+                                 'has a different category than in the '
+                                 'product form.')}
+                self.product_uom_id = self.product_id.uom_id
+
+
+class Product(models.Model):
+    _inherit = 'product.product'
+
+    business_usage_ids = fields.One2many('business.product.line', 'product_id',
+                                         'Business Usage')
+
+
+class StockLocation(models.Model):
+    _inherit = 'stock.location'
+
+    business_usage_id = fields.Many2one('business.product.location',
+                                        'Business Usage')
+
+
+class BusinessProductLocationReport(models.AbstractModel):
```

Comment:
According to the OCA guidelines, reports shoudl be in a report module. https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#directories